### PR TITLE
Decay functions and Generalized Costs

### PR DESCRIPTION
**Pull request by abyrd**
This adds documentation for two new features slated for public release: distance decay functions and generalized cost from specialized OSM tags.

It also fixes a Google Analytics problem with non-production deployments of the docs.

This is marked draft because it needs another round of cleanup and writing.

### CODE REVIEW

File: docs/learn-more/generalized-cost.mdx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+There is currently no technical difference between the way more accurate timings and generalized costs are handled. The only difference is in the interpretation. In either case, the plugins return adjusted distances (which are traversed at the specified speeds) and turning times (which are constants unaffected by speed).
+
+The length of every street segment between intersections can be scaled independently. At every intersection, each left and right turn and each movement straight through an intersection can have a distinct cost.
+
+## LADOT Generalized Costs
+
+Specifically in the LADOT case, we use cost supplier plugins (LaDotBikeCostSupplier, LaDotWalkCostSupplier) which use OSM tags read by the LaDotCostTags class. That input OSM data must be processed into ways that have no intersections except at their endpoints. All the required tags must be present on every edge. If not, the generalized cost functionality will be distabled. OSM data in this form is created by the data preparation scripts at https://github.com/RSGInc/ladot_analysis_dataprep. The Python OSM library used there naturally breaks ways into routable edges between intersections as required.
```
Comment (ansoncfit) [inline docs/learn-more/generalized-cost.mdx]:
```suggestion
Specifically in the LADOT case, we use cost supplier plugins (LaDotBikeCostSupplier, LaDotWalkCostSupplier) which use OSM tags read by the LaDotCostTags class. That input OSM data must be processed into ways that have no intersections except at their endpoints. All the required tags must be present on every edge. If not, the generalized cost functionality will be disabled. OSM data in this form is created by the data preparation scripts at https://github.com/RSGInc/ladot_analysis_dataprep. The Python OSM library used there naturally breaks ways into routable edges between intersections as required.
```

File: docs/learn-more/generalized-cost.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- speed_peak
+- speed_offpeak 
+
+### Walk Costs
+
+This explains how walk costs are derived from the tags. 
+
+Perceived length: sections over 10% grade count double. Unpaved roads and alleys count 51% more. Busy roads are penalized by 14%.
+
+Turn costs are defined in terms of meters traveled at a standardized speed of 1.3 meters per second (not affected by the walking speed in the analysis request). Traversing any intersection has a fixed cost of 54 meters. Daily traffic over 10k adds a penalty of 28 meters if there is no crosswalk at all; daily traffic over 13k adds a penalty of 73 meters if there is no signalized crosswalk. The traffic figure used is the cross traffic when moving straight through an intersection; it is the greater of the traffic in the forward or backward directions when turning left (across traffic).
```
Comment (ansoncfit) [inline docs/learn-more/generalized-cost.mdx]:
```suggestion
Turn costs are defined in terms of meters traveled at a standardized speed of 1.3 meters per second (not affected by the walking speed in the analysis request). Traversing any intersection has a fixed cost of 54 meters (41.5 seconds). Daily traffic over 10k adds a penalty of 28 meters (21.5 seconds) if there is no crosswalk at all; daily traffic over 13k adds a penalty of 73 meters (56.2 seconds) if there is no signalized crosswalk. The traffic figure used is the cross traffic when moving straight through an intersection; it is the greater of the traffic in the forward or backward directions when turning left (across traffic). 
```

File: docs/learn-more/generalized-cost.mdx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+### Bike Costs
+
+This explains how bike costs are derived from the tags.
+
+Perceived length: sections at 2-4% grade count 37% more, sections at 4-6% grade count 123% more, and those over 6% grade count 324% more. Bike boulevards get an 11% bonus and bike paths a 16% bonus. Where bike infrastructure is not present, busy roads are penalized based on the road's average daily traffic. Roads over 30k vehicles/day are penalized by 716%, those over 20k by 140% and those over 10k by 37%.
+
+Turn costs are defined in terms of meters traveled at a standardized biking speed of 4 meters per second (not affected by the biking speed in the analysis request). All turns receive a penalty of 54 meters (which does not apply when going straight through an intersection). Stop signs and signals add 6 and 27 meters respectively.
```
Comment (ansoncfit) [inline docs/learn-more/generalized-cost.mdx]:
```suggestion
Turn costs are defined in terms of meters traveled at a standardized biking speed of 4 meters per second (not affected by the biking speed in the analysis request). All turns receive a penalty of 54 meters (13.5 seconds), which does not apply when going straight through an intersection. Stop signs and signals add 6 and 27 meters (1.5 and 6.8 seconds) respectively. 
```

File: docs/learn-more/generalized-cost.mdx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+### Bike Costs
+
+This explains how bike costs are derived from the tags.
+
+Perceived length: sections at 2-4% grade count 37% more, sections at 4-6% grade count 123% more, and those over 6% grade count 324% more. Bike boulevards get an 11% bonus and bike paths a 16% bonus. Where bike infrastructure is not present, busy roads are penalized based on the road's average daily traffic. Roads over 30k vehicles/day are penalized by 716%, those over 20k by 140% and those over 10k by 37%.
+
+Turn costs are defined in terms of meters traveled at a standardized biking speed of 4 meters per second (not affected by the biking speed in the analysis request). All turns receive a penalty of 54 meters (which does not apply when going straight through an intersection). Stop signs and signals add 6 and 27 meters respectively. 
+
+Right turns are penalized by 50 meters if the daily traffic of the cross road exceeds 10k. Left turns onto busy cross streets, and passing straight through intersections with busy streets are penalized in the same way: 424 meters if the cross traffic exceeds 20k/day, 81 meters if it exceeds 10k/day, and 78 meters if it exceeds 5k/day.
```
Comment (ansoncfit) [inline docs/learn-more/generalized-cost.mdx]:
```suggestion
Right turns are penalized by 50 meters (12.5 seconds) if the daily traffic of the cross road exceeds 10k. Left turns onto busy cross streets, and passing straight through intersections with busy streets are penalized in the same way: 424 meters (1 minute, 46 seconds) if the cross traffic exceeds 20k/day, 81 meters (20.3 seconds) if it exceeds 10k/day, and 78 meters (19.5 seconds) if it exceeds 5k/day.
```

File: docs/learn-more/generalized-cost.mdx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+Left turns from busy streets are also affected by parallel traffic: 297 meters for over 20k/day, and 117 meters for over 10k/day.
```
Comment (ansoncfit) [inline docs/learn-more/generalized-cost.mdx]:
```suggestion
Left turns from busy streets are also affected by parallel traffic: 297 meters (1 minute, 14 seconds) for over 20k/day, and 117 meters (29.3 seconds) for over 10k/day.
```

**Review by ansoncfit — CHANGES_REQUESTED** [review]
These new sections look good overall. My main comment is that the generalized cost introduction refers to "turning times," so we should specify times in the discussion of the turn penalties.

**Comment by abyrd** [comment]
I did some more fact-checking and cleanup to match recent backend changes. This should be ready for review and potentially Kodiak automerge @ansoncfit

**Review by ansoncfit — APPROVED** [review]
👍

**Pull request merged**